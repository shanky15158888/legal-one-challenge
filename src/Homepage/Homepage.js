import React from 'react'
import Table from '../Table/Table'
import agents from '../json-data/agents.json'
import logs from '../json-data/logs.json'
import moment from 'moment'

export default function Homepage(props) {
    const cols = ["Phone number", "Number of calls", "Last call details"]
    const tableData = [];
    const formatBasedOnNumber = getFormatBasedOnNumber();
    for(let key in formatBasedOnNumber) {
        const sortedRecord = sortRecordBasedOnDate(formatBasedOnNumber[key]);
        const obj = {}
        obj['data'] = []
        obj.data.push({'url': `/call/${sortedRecord[0].number}`, 'type': 'link', 'data': sortedRecord[0].number})                
        obj.data.push({'url': '', 'data': `${sortedRecord.length} ${sortedRecord.length > 1? "calls": "call"}`})                
        obj.data.push({'url': `/agent/${sortedRecord[0].agentIdentifier}`, 'type': 'link', 'data': `${sortedRecord[0].name}/${getTime(sortedRecord[0].dateTime)}`})                
        tableData.push(obj)
    }
    return (
        <>
            <header>
                <h2>Homepage</h2>
            </header>
            <Table cols={cols} rows={tableData} />
        </>
    )
}

const sortRecordBasedOnDate = (arr) => {
    const list = [...arr]
    list.sort((a, b) => {
        return Date.parse(b.dateTime) - Date.parse(a.dateTime)
    })
    return list;
}

const getTime = (date) => {
    return moment(date).format('HH:MM');
}

const getFormatBasedOnNumber = () => {
    let formatBasedOnNumber = {};
    logs.forEach((log) => {
        const agentIdentifier = formatBasedOnNumber[log.number];
        agents.forEach((agentDetail) => {
            if(agentDetail['identifier'] === log.agentIdentifier) {
                log['name'] = `${agentDetail['firstName']} ${agentDetail['lastName']}`;
            } 
        })
        if(agentIdentifier) {
            formatBasedOnNumber[log.number].push(log)
        } else {
            formatBasedOnNumber[log.number] = [log]

        }
    })
    return formatBasedOnNumber
}


