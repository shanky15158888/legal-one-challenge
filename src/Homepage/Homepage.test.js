import React from 'react';
import renderer from 'react-test-renderer';
import Homepage from './Homepage'
import { BrowserRouter as Router } from "react-router-dom";

test('Homepage component testing', () => {
    const component = renderer.create(
        <Router>
            <Homepage />
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});