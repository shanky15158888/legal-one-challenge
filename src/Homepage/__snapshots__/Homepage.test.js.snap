// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component testing 1`] = `
Array [
  <header>
    <h2>
      Homepage
    </h2>
  </header>,
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Phone number
        </th>
        <th
          scope="col"
        >
          Number of calls
        </th>
        <th
          scope="col"
        >
          Last call details
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        scope="col"
      >
        <td>
          <a
            href="/call/+49151484522"
            onClick={[Function]}
          >
            +49151484522
          </a>
        </td>
        <td>
          13 calls
        </td>
        <td>
          <a
            href="/agent/356b03dc-9ec5-11e7-97a6-d501104f897e"
            onClick={[Function]}
          >
            Abraham Ellis/20:10
          </a>
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          <a
            href="/call/+49158544147"
            onClick={[Function]}
          >
            +49158544147
          </a>
        </td>
        <td>
          3 calls
        </td>
        <td>
          <a
            href="/agent/e512594e-a34c-11e7-a6cb-0609e42722e2"
            onClick={[Function]}
          >
            Joel Wagner/23:11
          </a>
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          <a
            href="/call/+49151783331"
            onClick={[Function]}
          >
            +49151783331
          </a>
        </td>
        <td>
          3 calls
        </td>
        <td>
          <a
            href="/agent/6fe55f02-a8f3-11e7-b037-c33c31bca71f"
            onClick={[Function]}
          >
            Madeline Lee/17:01
          </a>
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          <a
            href="/call/+49151514231"
            onClick={[Function]}
          >
            +49151514231
          </a>
        </td>
        <td>
          3 calls
        </td>
        <td>
          <a
            href="/agent/6fe55f02-a8f3-11e7-b037-c33c31bca71f"
            onClick={[Function]}
          >
            Madeline Lee/23:02
          </a>
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          <a
            href="/call/+49221514231"
            onClick={[Function]}
          >
            +49221514231
          </a>
        </td>
        <td>
          6 calls
        </td>
        <td>
          <a
            href="/agent/6fe55f02-a8f3-11e7-b037-c33c31bca71f"
            onClick={[Function]}
          >
            Madeline Lee/20:09
          </a>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
